code/game/machinery/computer/buildandrepair.dm

```diff
@@ -488,16 +489,21 @@
 				new /obj/item/stack/cable_coil(get_turf(src), 5)
 				return 1
 
-			if(istype(P, /obj/item/stack/sheet/glass/glass))
-				var/obj/item/stack/sheet/glass/glass/G = P
+			if(istype(P, /obj/item/stack/sheet/glass))
+				var/obj/item/stack/sheet/glass/G = P
+				if(istype(G, /obj/item/stack/sheet/glass/rglass) || istype(G, /obj/item/stack/sheet/glass/plasmarglass)) // Don't use these
+					to_chat(user, "<span class='warning'>Sheets of glass must not have rods in them!</span>")
+					return
 				if (G.amount < 2)
 					to_chat(user, "<span class='warning'>You need at least 2 sheets of glass for this!</span>")
 					return 1
 				to_chat(user, "<span class='notice'>You start installing the glass panel onto the frame.")
 				if(do_after(user, src, 20) && state == 3 && G.amount >= 2)
 					playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
 					G.use(2)
-					user.visible_message("[user] installs the glass panel onto the frame.", "You install the glass panel onto the frame.", "You hear metallic sounds.")
+					user.visible_message("[user] installs the [empproof ? "plasma" : ""] glass panel onto the frame.", "You install the [empproof ? "plasma" : ""] glass panel onto the frame.", "You hear metallic sounds.")	
```

Comment:
This is very confusing. It seems that if you install a fresh glass panel into the computer that was plasmaglass, it will say you install plasmaglass because the computer itself is EMP-proof.

empproof isn't set until later though so if you use plasmaglass on a computer it will say it's normal glass

Reply:
Moved the message after the variable setting